cms/admin/pageadmin.py

```diff
@@ -540,36 +520,40 @@ def has_add_permission(self, request):
         """
         Return true if the current user has permission to add a new page.
         """
-        if get_cms_setting('PERMISSION'):
+        can_add = super(PageAdmin, self).has_add_permission(request)
+
+        if can_add and get_cms_setting('PERMISSION'):
```

Comment:
Does it make sense to put something like: `CMS_PERMISSION = get_cms_setting('PERMISSION')` at the top of this file and then use the "constant": `CMS_PERMISSION` everywhere instead? Its unlikely to change between requests =)